### DISCUSSION TITLE
WSL is really not integrated well

### DISCUSSION THREAD

**Discussion by nyck33** [microsoft/WSL#9109] [General]
### Version

n

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5

### Distro Version

bb

### Other Software

b

### Repro Steps

b

### Expected Behavior

b

### Actual Behavior

There should be just one file system, the Windows one.  
Put everything under the hood to make is seamless.  
I know you don't like how I only partially filled out the boxes but I feel like that's what I got as a Linux Subsystem.  Something that's incomplete.

### Diagnostic Logs

b

**Comment by NotTheDr01ds** [comment]
> There should be just one file system, the Windows one.

If you have a serious and rational proposal on how to implement that, you can certainly make a feature request, but this is not an "issue" with WSL.  WSL runs most any Linux distribution, and Linux distributions need a *Linux*-compatible filesystem that they can understand and work with.  NTFS (the "Windows one") is not fully POSIX/Linux compatible.

* Windows needs a Windows-compatible filesystem
* Linux needs a Linux-compatible filesystem

WSL makes a (IMHO) reasonable compromise by making a Linux-compatible filesystem available to the Linux distributions running in WSL.  This provides the greatest level of compatibility for Linux applications.  Without it, many Linux applications would have issues or fail outright when running in WSL.

WSL also provides integration between the two disparate types of filesystems by:

* Making the Windows filesystem available to Linux distributions by default using `/mnt/c` (etc.) automounts.
* Making the Linux filesystem available to Windows through `\\wsl$\<distribution>` (Windows 10/11) or the preferred `\\wsl.localhost\<distribution>` (Windows 11)

This is a far higher level of integration than another similar environment like a VM would provide out-of-the-box.

If you would, however, prefer to run everything on the Windows filesystem, you might consider [Cygwin](https://www.cygwin.com), but realize that everything there must be compiled to run under it.  It won't run native Linux ELF binaries.

You might also consider a "Linux-like" shell that can run under Windows.  Git Bash is one, but there are many others.  The fairly new [Nushell](https://www.nushell.sh), as just one example, is available for both Linux/Unix and Windows OSs.

**Comment by nyck33** [comment]
Nope.
```
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/nobut$ ls -al
ls: cannot open directory '.': Permission denied
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/nobut$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ ls
'All Users'  'Default User'   defaultuser100000   desktop.ini     nobut
 Default      Public          defaultuser100001   nobuk_53lode3
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ cd Public
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ ls
AccountPictures  Desktop  Documents  Downloads  Libraries  Music  Pictures  Videos  desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ ls
AccountPictures  Desktop  Documents  Downloads  Libraries  Music  Pictures  Videos  desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Downloads
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Downloads$ ls
desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Downloads$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Desktop
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Desktop$ ls
 Firefox.lnk  'Google Chrome.lnk'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Desktop$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ pwd
/mnt/c/Users/Public
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ ls
'All Users'  'Default User'   defaultuser100000   desktop.ini     nobut
 Default      Public          defaultuser100001   nobuk_53lode3
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ cd 'Default User'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ ls
 AppData
'Application Data'
 Cookies
 Desktop
 Documents
 Downloads
 Favorites
 Links
'Local Settings'
 Music
'My Documents'
 NTUSER.DAT
 NTUSER.DAT.LOG1
 NTUSER.DAT.LOG2
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TM.blf
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000001.regtrans-ms
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000002.regtrans-ms
 NetHood
 Pictures
 PrintHood
 Recent
'Saved Games'
 SendTo
 Templates
 Videos
'スタート メニュー'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User/Documents$ ls
'My Music'  'My Pictures'  'My Videos'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ ls
 AppData
'Application Data'
 Cookies
 Desktop
 Documents
 Downloads
 Favorites
 Links
'Local Settings'
 Music
'My Documents'
 NTUSER.DAT
 NTUSER.DAT.LOG1
 NTUSER.DAT.LOG2
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TM.blf
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000001.regtrans-ms
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000002.regtrans-ms
 NetHood
 Pictures
 PrintHood
 Recent
'Saved Games'
 SendTo
 Templates
 Videos
'スタート メニュー'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ ls
'All Users'  'Default User'   defaultuser100000   desktop.ini     nobut
 Default      Public          defaultuser100001   nobuk_53lode3
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ cd 'All Users'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users$ ls
 A-Volute             'Microsoft Visual Studio'                       USOPrivate
'Application Data'     Mozilla-1de4eec8-1241-4177-a864-e594e8d1fb38   USOShared
 Documents             NVIDIA                                         WindowsHolographicDevices
 Lenovo               'NVIDIA Corporation'                            regid.1991-06.com.microsoft
 Logishrd             'Package Cache'                                 ssh
 McAfee                Packages                                      'スタート メニュー'
 Microsoft             SoftwareDistribution                           デスクトップ
'Microsoft OneDrive'   Templates
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users$ ls
 A-Volute             'Microsoft Visual Studio'                       USOPrivate
'Application Data'     Mozilla-1de4eec8-1241-4177-a864-e594e8d1fb38   USOShared
 Documents             NVIDIA                                         WindowsHolographicDevices
 Lenovo               'NVIDIA Corporation'                            regid.1991-06.com.microsoft
 Logishrd             'Package Cache'                                 ssh
 McAfee                Packages                                      'スタート メニュー'
 Microsoft             SoftwareDistribution                           デスクトップ
'Microsoft OneDrive'   Templates
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/All Users$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ ls
'All Users'  'Default User'   defaultuser100000   desktop.ini     nobut
 Default      Public          defaultuser100001   nobuk_53lode3
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ cd Public
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ ls
AccountPictures  Desktop  Documents  Downloads  Libraries  Music  Pictures  Videos  desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ ls
'My Music'  'My Pictures'  'My Videos'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd Desktop
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Desktop$ ls
 Firefox.lnk  'Google Chrome.lnk'   desktop.ini
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public/Desktop$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Public$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ ls
'All Users'  'Default User'   defaultuser100000   desktop.ini     nobut
 Default      Public          defaultuser100001   nobuk_53lode3
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users$ cd 'Default User'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ ls
 AppData
'Application Data'
 Cookies
 Desktop
 Documents
 Downloads
 Favorites
 Links
'Local Settings'
 Music
'My Documents'
 NTUSER.DAT
 NTUSER.DAT.LOG1
 NTUSER.DAT.LOG2
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TM.blf
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000001.regtrans-ms
 NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000002.regtrans-ms
 NetHood
 Pictures
 PrintHood
 Recent
'Saved Games'
 SendTo
 Templates
 Videos
'スタート メニュー'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$ cd Documents
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User/Documents$ ls
'My Music'  'My Pictures'  'My Videos'
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User/Documents$ cd ..
nobu@LAPTOP-DAGJCC8T:/mnt/c/Users/Default User$
```
But two folders should be there.
![image](https://user-images.githubusercontent.com/25189545/199538827-8c47898b-8d1e-4bf0-9069-91f21f82784a.png)

**Comment by nyck33** [comment]
I don't even have Music, Pictures, Videos in Documents in Windows.  Where are these coming from?
`wsl --install` did not work on this Lenovo Gaming Ideapad bought in Japan so I had to go into Windows store, install from there, worked on the third try or so.  Why am I getting permission denied?  
I spent an entire day trying to work this out.  Can you please put some caveats like, "We made this but it doesn't really work" next time?

**Comment by nyck33** [comment]
You need a more diverse team.  What you claim to be a reasonable compromise is totally unreasonable by standards outside of your team or Microsoft.

**Comment by NotTheDr01ds** [comment]
@nyck33  You seem to be addressing me as if I work for Microsoft (which I do not), since you are quoting my "reasonable compromise" comment.  It appears that you've been around Github a while, so I hope you realize that members of a repo have a designation next to their posts ("Owner", "Member", "Collaborator", etc.).  I'm just a general member of the WSL community who was trying to help you out (both here and on [Ask Ubuntu](https://askubuntu.com/q/1438681/1165986)).

> Can you please put some caveats like, "We made this but it doesn't really work" next time?

While I understand that you are frustrated, your wording here continues to come across as highly aggressive.

You also seem to not want to consider that the situation may be due to errors *you* made during installing.  I'm not saying this is absolutely the case, but as I pointed out in a comment on Ask Ubuntu, [the site](https://www.sitepoint.com/wsl2/) that you say that you followed contains a mixture of outdated and simply *wrong* instructions.  I am not at all surprised that you had multiple issues while installing, and that your system may be left in a bad state as a result.

I'm sorry that's the case for you, but the WSL team has made installation incredibly easy and [quite well-documented](https://learn.microsoft.com/en-us/windows/wsl/install).

It also appears to me that there's a significant possibility that your system was *already* in a bad state when you bought it.  There are at least 6 users on your system, and it seems that the WSL distribution itself may have been installed as one of those *other* users.  If you did not intentionally create those users, I would highly recommend that you wipe the system and reinstall Windows to make sure that you return things to a stable and secure state.
